File: Java/Medium/1508.java
class Solution {
    public int rangeSum(int[] nums, int n, int left, int right) {
        // brute force
        // use a new array to store the sum of subarrays
        // create the new array
        ArrayList<Integer> temp = new ArrayList<>();

        // fill the new array
        for (int i = 0; i < n; i++) {
            int sum = 0;
            for (int j = i; j < n; j++) {
                sum += nums[j];
                temp.add(sum);
            }
        }

        // sort the array
        Collections.sort(temp);

        int answer = 0;
        int mod = (int) Math.pow(10, 9) + 7;
        // go through the new array to get the answer
        for (int k = left - 1; k < right; k++) {
            answer = (answer + temp.get(k)) % mod;
        }

        return answer;
    }
}

File: Python/Medium/1508.py
class Solution:
    def rangeSum(self, nums: List[int], n: int, left: int, right: int) -> int:
        # brute force
        # use a new temp array to store the subarray sums
        temp = []

        # fill the array
        for i in range(n):
            sum = 0
            for j in range(i, n):
                sum += nums[j]
                temp.append(sum)

        # sort the temp array
        temp.sort()

        # go through the array to get the answer
        answer = 0
        mod = 10**9 + 7
        for i in range(left - 1, right):
            answer = (answer + temp[i]) % mod

        return answer
